packages/common/src/ui/components/buttons/standard/ButtonWithIcon.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { ButtonProps, Tooltip } from '@mui/material';
+import { LocaleKey, useTranslation } from '../../../../intl/intlHelpers';
+import { useIsSmallScreen } from '../../../../hooks';
+import { ShrinkableBaseButton } from './ShrinkableBaseButton';
+
+interface ButtonWithIconProps extends ButtonProps {
+  Icon: React.ReactNode;
+  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  labelKey: LocaleKey;
+  shouldShrink?: boolean;
+  variant?: 'outlined' | 'contained';
+  color?: 'primary' | 'secondary';
+  disabled?: boolean;
+}
+
+export const ButtonWithIcon: React.FC<ButtonWithIconProps> = ({
+  labelKey,
+  onClick,
+  Icon,
+  shouldShrink = true,
+  variant = 'outlined',
+  color = 'primary',
+  disabled,
+  ...buttonProps
+}) => {
+  const t = useTranslation();
+  const isSmallScreen = useIsSmallScreen();
+
+  // On small screens, if the button shouldShrink, then
+  // only display a centered icon, with no text.
+  const shrink = isSmallScreen && shouldShrink;
+  const startIcon = shrink ? null : Icon;
+  const centeredIcon = shrink ? Icon : null;
+  const text = shrink ? null : t(labelKey);
```

Comment:
I shifted this here because it didn't make sense to have a shrinkable button that wasn't an icon button?